File: scrimage-core/src/test/scala/com/sksamuel/scrimage/metadata/ImageMetadataTest.scala
package com.sksamuel.scrimage.metadata

import com.sksamuel.scrimage.{ImageMetadata, Tag}
import org.scalatest.{Matchers, WordSpec}

class ImageMetadataTest extends WordSpec with Matchers {

  private val stream = getClass.getResourceAsStream("/vossen.jpg")

  "metadata" should {
    "read EXIF" in {
      val meta = ImageMetadata.fromStream(stream)
      meta.tags should contain(new Tag("ISO Speed Ratings", 34855, "2500", "2500"))
      meta.tags should contain(new Tag("Image Width", 256, "4928", "4928 pixels"))
      meta.tags should contain(new Tag("White Balance Mode", 41987, "0", "Auto white balance"))
    }
  }
}

File: scrimage-core/src/test/kotlin/com/sksamuel/scrimage/core/metadata/ImageMetadataTest.kt
@file:Suppress("BlockingMethodInNonBlockingContext")

package com.sksamuel.scrimage.core.metadata

import com.sksamuel.scrimage.metadata.ImageMetadata
import com.sksamuel.scrimage.metadata.Tag
import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.collections.shouldContain

class ImageMetadataTest : WordSpec({

   val stream = javaClass.getResourceAsStream("/vossen.jpg")

   "metadata" should {
      "read EXIF" {
         val meta = ImageMetadata.fromStream(stream)
         meta.tags().toList().shouldContain(
            Tag(
               "ISO Speed Ratings",
               34855,
               "2500",
               "2500"
            )
         )
         meta.tags().toList().shouldContain(
            Tag(
               "Image Width",
               256,
               "4928",
               "4928 pixels"
            )
         )
         meta.tags().toList().shouldContain(
            Tag(
               "White Balance Mode",
               41987,
               "0",
               "Auto white balance"
            )
         )
      }
   }

})
